### PR TITLE
[expo-file-system][next] Add `.list` to expo-file-system/next

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Adding `.list` to a `directory` class to get directory contents.

# How

Needed to use some JS code for now.

# Test Plan

Added unit tests

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-file-system/src/next/FileSystem.types.ts [hunk 1]
```diff
@@ -52,6 +52,15 @@ export declare class Directory {
    * Moves a directory.
    */
   move(destination: Directory | File);
+  /**
+   * @hidden
+   * Lists the contents of a directory. Should not be used directly, as it retuns a list of paths.
```
Comment (lukmccall) [inline packages/expo-file-system/src/next/FileSystem.types.ts:57]:
```suggestion
   * Lists the contents of a directory. Should not be used directly, as it returns a list of paths.
```

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemNextModule.kt [hunk 2]
```diff
@@ -130,6 +130,10 @@ class FileSystemNextModule : Module() {
       Property("path") { directory ->
         directory.asString()
       }
+
+      Function("listAsRecords") { directory: FileSystemDirectory ->
```
Comment (lukmccall) [inline packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemNextModule.kt:134]: I don't know if I like this name or not. Let me think about a better one.

**Review by lukmccall — APPROVED** [review]

File: packages/expo-file-system/ios/Next/FileSystemDirectory.swift [hunk 3]
```diff
@@ -15,6 +15,16 @@ internal final class FileSystemDirectory: FileSystemPath {
...
     return false
   }
+
+  func listAsRecords() throws -> [[String: Any]] {
+    var contents: [[String: Any]] = []
+
+    let items = try FileManager.default.contentsOfDirectory(at: url, includingPropertiesForKeys: nil)
+      for item in items {
+        contents.append(["isDirectory": item.hasDirectoryPath, "path": item.absoluteString])
+      }
```
Comment (tsapeta) [inline packages/expo-file-system/ios/Next/FileSystemDirectory.swift:25]:
```suggestion
    for item in items {
      contents.append(["isDirectory": item.hasDirectoryPath, "path": item.absoluteString])
    }
```

File: packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemNextModule.kt (hunk 2 above)
Comment (tsapeta) [reply packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemNextModule.kt:147]: Have you found a better one? 😄 If not, could you @aleqsio add a comment (on Android, iOS and TS) that this function is internal and will be removed in the future (when returning shared objects is supported).

**Review by tsapeta — APPROVED** [review]

**Pull request merged**